Create BYOC cluster GCP (#995)
redpanda-data/documentation: docs/cloud/create-byoc-cluster-gcp.mdx

## Create a namespace

To prepare to deploy a Redpanda cluster in your own cloud, log in to the Redpanda Cloud UI and add a namespace:

1. Go to cloudv2.repanda.com and click **Log in**.

    The Enter Organization page opens.
2. Type your organization name and click **Continue**.

    The Home page opens.
3. Click **Add namespace**.

    Namespaces help you organize clusters by owner, team, or project for example.
4. Type a name for your namespace.

## Create a BYOC cluster

After you create a namespace, create a BYOC cluster in your Virtual Private Cloud (VPC):

1. Scroll to your namespace (namespaces are listed in alphabetical order) and click **+ Add Cluster**.
2. Click **Create BYOC cluster**.
3. (Optional) In **Cluster name**, type a name to replace the default name.
4. Select throughput.

    The choices are 1 MBps, 5 MBps, 10 MBps, and 20 MBps.
5. Open the **Redpanda version** menu and select a version.
6. Click **Next**.

    The **Cloud settings** page opens.
7. For **Provider**, select GCP.
8. Open the **Region** menu and select a region.

> 9. For **Availability**, select **Single AZ** or **Multi AZ**

Reviewer: ```suggestion
9. For **Availability**, select **Single AZ** or **Multi AZ**.
```

10. Open the **Zones** menu and select the availability zone(s) for your region.

>     If Single AZ was selected in the previous step then select a single zone. If Multi AZ was selected in the previous step then select three availability zones to provide two backups in case one availability zone goes down.

Reviewer: ```suggestion
    If Single AZ was selected in the previous step, then select a single zone. If Multi AZ was selected, then select three availability zones to provide two backups in case one availability zone goes down.
```

11. Click **Next**.

    The **Networking** page opens.
12. Under **Choose Connection Type**, select either **Public** or **Private**.
13. Click **Next**.

After the network is created, you can deploy the agent.

> ## Deploying the agent

Reviewer: ```suggestion
## Deploy the agent
```

> ### Prepare deployment environment (prerequisites)

Reviewer: ```suggestion
### Prerequisites
```

* Install the [Redpanda CLI](https://docs.redpanda.com/docs/platform/quickstart/rpk-install/).
    * Minimum `rpk` version: `v22.3+`.
* Install the [gcloud CLI](https://cloud.google.com/sdk/docs/install).
    * If the gcloud CLI is already installed, make sure the components are up-to-date by running: `gcloud components update`.
* Setup the gcloud [application-default](https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login) credentials.

>     To temporarily use your user credentials for API access run: `gcloud auth application-default login`. This command will open Google's sign in page in a browser window.

Reviewer: ```suggestion
    To temporarily use your user credentials for API access, run: `gcloud auth application-default login`. This command opens Google's log in page in a browser window.
```

* Ensure that your role has the necessary permissions to run commands and create clusters.

### Create Redpanda Cloud client

* From the Redpanda Cloud home page, select **Clients** from the navigation pane.
* Click **Add client** and enter a name and description, or simply use an existing client.

>     Copy the **Client ID** and **Client Secret**

Reviewer: ```suggestion
    Copy the **Client ID** and **Client Secret**.
```

> * Navigate back to the cluster and the wizard will take you back to the deploy agent page.

Reviewer: ```suggestion
* Navigate back to the cluster. The wizard takes you back to the Deploy Agent page.
```

>     Paste the **Client ID** and **Client Secret** in to the client credentials boxes.

Reviewer: ```suggestion
    Paste the **Client ID** and **Client Secret** in the client credentials boxes.
```

* Run the `rpk cloud login` command in your deployment environment:

```shell
rpk cloud login --save 
  --client-id='...'
  --client-secret='...'
```

### Deploy agent

> * Enter your GCP Project ID

Reviewer: ```suggestion
* Enter your GCP Project ID.
```
